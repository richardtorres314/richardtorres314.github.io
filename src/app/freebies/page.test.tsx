import "@testing-library/jest-dom";

import { render, screen } from "@testing-library/react";

import Page from "./page";

describe("Freebies Page", () => {
  it("renders freebies page unchanged", () => {
    const { container } = render(<Page />);
    expect(container).toMatchSnapshot();
  });

  it("renders freebies list", () => {
    render(<Page />);
    const freebiesList = screen.getByRole("list").childElementCount;
    expect(freebiesList).toBeGreaterThan(0);
  });
});
