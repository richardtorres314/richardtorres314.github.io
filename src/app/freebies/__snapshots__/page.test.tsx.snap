// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Freebies Page renders freebies page unchanged 1`] = `
<div>
  <div
    class="fade css-0"
    style="opacity: 0; animation-delay: 0ms;"
  >
    <section
      class="p-12 scroll-mt-24 print:break-inside-avoid print:p-0"
    >
      <div
        class="container"
      >
        <div
          class="mb-16 text-center"
        >
          <h1
            class="text-4xl font-bold mb-6 text-center"
          >
            <span
              class="text-red-500"
            >
              Fre
            </span>
            ebies
          </h1>
          <p
            class="mb-4"
          >
            The best things in life are free!
          </p>
          <p>
            Here is a collection of free resources created by me you can use in your next project.
          </p>
        </div>
        <ul>
          <li>
            <a
              href="https://gist.github.com/richardtorres314/26b18e12958ba418bb37993fdcbfc1bd"
              rel="noreferrer"
              target="_blank"
            >
              <div
                class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm hover:bg-card-foreground/5"
                data-slot="card"
              >
                <div
                  class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 px-6 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6"
                  data-slot="card-header"
                >
                  <div
                    class="leading-none font-semibold"
                    data-slot="card-title"
                  >
                    Flexbox SASS Mixin
                  </div>
                </div>
                <div
                  class="px-6"
                  data-slot="card-content"
                >
                  <div
                    class="flex flex-wrap gap-2 mb-4"
                  >
                    <div
                      class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-hidden focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-primary text-primary-foreground hover:bg-primary/80"
                    >
                      SASS
                    </div>
                  </div>
                  <p>
                    A collection of mixins for flexbox usage in SASS
                  </p>
                </div>
              </div>
            </a>
          </li>
        </ul>
      </div>
    </section>
  </div>
</div>
`;
